### PR TITLE
Add query parameters to create-app page for pre-filling form

### PR DESCRIPTION
**Pull request by nenb**
## Reference Issues or PRs

Issue: https://github.com/nebari-dev/jhub-apps/issues/472

## What does this implement/fix?

_Put a `x` in the boxes that apply_

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds a feature)
- [ ] Breaking change (fix or feature that would cause existing features not to work as expected)
- [ ] Documentation Update
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build related changes
- [ ] Other (please describe):

## Testing

- [x] Did you test the pull request locally?
- [x] Did you add new tests?

#### Text styling

- [ ] The content is written with [plain language](https://www.plainlanguage.gov/guidelines/) (where relevant).
- [ ] If there are headers, they use the proper header tags (with only one level-one header: `H1` or `#` in markdown).
- [ ] All links describe where they link to (for example, check the [Nebari website](https://nebari.dev/)).
- [ ] This content adheres to the Nebari style guides.

#### Non-text content

- [ ] All content is represented as text (for example, images need alt text, and videos need captions or descriptive transcripts).
- [ ] If there are emojis, there are not more than three in a row.
- [ ] Don't use [flashing GIFs or videos](https://www.w3.org/TR/UNDERSTANDING-WCAG20/seizure-does-not-violate.html).
- [ ] If the content were to be read as plain text, it still makes sense, and no information is missing.

### CODE REVIEW

File: ui/src/App.tsx [hunk 1]
```diff
@@ -30,6 +31,7 @@ export const App = (): React.ReactElement => {
   const [notification, setNotification] = useRecoilState<string | undefined>(
     currentNotification,
   );
+  const [searchParams] = useSearchParams();
```
Comment (jbouder) [inline ui/src/App.tsx:34]: Unless you see this query param being passed to other page components as props, I don't think this should be in the top-level App Component. I would suggest moving it to the CreateApp Component...or even maybe the AppForm component itself if possible.

**Review by jbouder — CHANGES_REQUESTED** [review]
In general looks good, but might be able to simplify your implementation as mentioned in the comment.

File: ui/src/components/app-form/app-form.tsx [hunk 2]
```diff
@@ -62,6 +65,8 @@ export const AppForm = ({ id }: AppFormProps): React.ReactElement => {
   const firstErrorRef = useRef<HTMLInputElement | null>(null);
   const appInfoRef = useRef<HTMLDivElement | null>(null);
 
+  const initialFilepath = initialValues?.filepath || '';
```
Comment (jbouder) [inline ui/src/components/app-form/app-form.tsx:68]: A file path will likely include special characters, so any query params passed to the app likely will need to be URL encoded. So 2 things:
1. Ensure all query params are URL encoded from the source JupyterLab implementation
2. Ensure all query params are URL decoded here. Something like `decodeURIComponent`

**Review by jbouder — CHANGES_REQUESTED** [review]
One more thing...see the comment

File: ui/src/components/app-form/app-form.tsx (hunk 2 above)
Comment (nenb) [reply ui/src/components/app-form/app-form.tsx:68]: @jbouder For 1., please see https://github.com/nebari-dev/jupyterlab-jhub-apps/blob/0e0faf6ba6d7107fe44e4518c5ac90641bb6bf4b/src/index.ts#L54-L58.

I believe that I've addressed 2. in the latest commit.

**Comment by jbouder** [comment]
PR looks good now. Just need to resolve the merge conflict on the build. You do that, you’ll need to do a few things:
1. Merge main into your branch
2. Delete the local index.js to resolve any merge conflicts on that file
4. Perform an ‘npm install’ from the ui directory
5. Perform an ‘npm run build’ from the ui directory 
6. Commit the new build and push

File: ui/src/components/app-form/app-form.tsx (hunk 2 above)
Comment (jbouder) [reply ui/src/components/app-form/app-form.tsx]: Side note…if this should be loaded into an iframe, you should be able to add the following query string param now to make the form load “headless”: ‘headless=true’

**Comment by nenb** [comment]
> PR looks good now. Just need to resolve the merge conflict on the build. To do that, you’ll need to do a few things:

Ah, sorry, I was halfway through a rebase before I say this. I think it should be okay now.

**Review by jbouder — APPROVED** [review]
Looks good!

**Review by krassowski — APPROVED** [review]
:rocket:

**Pull request merged**